runner.go

```diff
@@ -258,20 +259,37 @@ func (r *runner) injectProcesses() bool {
 	return true
 }
 
-func (r *runner) inject(v namedInjectable) error {
-	r.logger.Info("Injecting services into %s", v.Name())
+func (r *runner) inject(injectable namedInjectable) error {
+	r.logger.Info("Injecting services into %s", injectable.Name())
 
-	if err := r.services.Inject(v.Wrapped()); err != nil {
+	if err := inject(r.services, r.logger, injectable); err != nil {
 		return fmt.Errorf(
 			"failed to inject services into %s (%s)",
-			v.Name(),
+			injectable.Name(),
 			err.Error(),
 		)
 	}
 
 	return nil
 }
 
+// inject will inject the given injectable with services. The service container
+// is first modified via overlay so that the logger is tagged with the service
+// name and any additional logging fields registered to the service.
+func inject(services service.ServiceContainer, logger log.Logger, injectable namedInjectable) error {
+	// Tag the logger with the service name and any registered log fields
+	logger = logger.WithFields(log.LogFields{"service": injectable.Name()})
```

Comment:
I'm not sure how I feel about this. I like the idea of it happening automatically but there's also not a way to remove it if someone doesn't want it (or wants it as a different name).

Reply:
You can supply something at registration time (see the loggingFields, which apply _over_ these fields). That would add an empty service string, though, not remove the key altogether (I think).

I'd be okay with removing this or having a combined convenience function instead. Thoughts on an alternate?

Reply:
I think leaving it up to the user is fine? The only reason I bring it up is in case someone wants to change the name from "service" to "initializer" or "process", or they just don't want it at all. Leaving it up to the user also makes it more discoverable if the users are looking through their logs and are wondering where a "service" field came from.

Reading this again, as another change I think it might help clarity to move the `injectable` to the beginning of the params since that's the object you're operating on.

Reply:
Ah! I didn't realize you meant configure the key and not the value. This makes sense!